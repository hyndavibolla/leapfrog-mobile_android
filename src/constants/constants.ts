export const pushNotificationTypes = {
  USER_INACTIVE: 'USER_INACTIVE',
  MISSION_COMPLETE: 'MISSION_COMPLETE',
  BALANCE_NEW_POINTS: 'BALANCE_NEW_POINTS',
  BALANCE_EXPIRING_POINTS: 'BALANCE_EXPIRING_POINTS',
  BALANCE_UNUSED_POINTS: 'BALANCE_UNUSED_POINTS',
  MARKETING: 'MARKETING',
  USER_BIRTHDAY: 'USER_BIRTHDAY'
};

export const RedemptionMethods = {
  IN_STORE_ONLY: 'In-Store only',
  ONLINE_ONLY: 'Online only',
  IN_STORE_OR_ONLINE_ONLY: 'In-Store and Online only',
  IN_RESTAURANT_ONLY: 'In Restaurant only',
  APP_ONLY: 'App only'
};

export const GROCERY_AND_DELIVERY_CATEGORY = 'Grocery & Delivery';

export const TutorialFrom = {
  TUTORIAL_REMINDER: 'tutorial_reminder',
  TUTORIAL_BANNER: 'tutorial_banner',
  PROFILE: 'profile'
};
