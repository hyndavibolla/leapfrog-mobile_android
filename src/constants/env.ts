// @ts-nocheck
import {
  ENVIRONMENT,
  BRANCH_NAME,
  API_URL,
  API_RETRY_REQUEST_ATTEMPTS,
  API_RETRY_REQUEST_ATTEMPT_DELAY_MS,
  API_RETRY_REQUEST_STATUS_CODE_RANGES_REGEX,
  SPLASH_MIN_ANIMATION_TIME_MS,
  SCHEME,
  WEBVIEWS_MAX_LOAD_ATTEMPTS,
  WEBVIEWS_ATTEMPT_DELAY_MS,
  SYW_URL,
  SYW_APP_PROFILE,
  SYW_SUPPORT_EMAIL,
  FUSION_APPLY_NOW_WEBVIEW_URI,
  FUSION_APPLICATION_FINISHED_URI_REGEX,
  API_TOKEN_EXPIRATION_PADDING_MS,
  IGNORED_FEATURE_LIST,
  LOCALE,
  TOAST_VISIBLE_MS,
  ACTIVITY_FETCH_HISTORY_MONTH_DELTA,
  LOCAL_OFFERS_DEFAULT_ACTIVATION_DAYS,
  LOCAL_OFFERS_ENROLLMENT_FINISH_URI_REGEX,
  MAX_ACTIVITY_FETCH_HISTORY_MONTH_DELTA,
  MERCHANT,
  MIN_ACTIVITY_FETCH_HISTORY_RESULTS,
  MISSION_KEEP_EARNING_LIMIT,
  MISSION_FETCH_LIMIT,
  CITI_LANDING_URL,
  CITI_PHONE_NUMBER,
  MODAL_PADDING_MS,
  API_CACHE_GAME_STATE_MS,
  API_CACHE_STREAK_LIST_MS,
  EARN_POINTS_ANIMATION_MS,
  SLIDE_URL,
  SLIDE_CLIENT_ID,
  SLIDE_CLIENT_SECRET,
  SLIDE_CHECKOUT_BASE_URL,
  SLIDE_CHECKOUT_PATH,
  SLIDE_CHECKOUT_CLIENT_HEADER,
  SLIDE_APPLICATION_ID,
  SLIDE_APPLICATION_FINISHED_URI_PREFIXES,
  GIFT_CARD_MAX_OPTIONS_COUNT,
  BUTTON_APP_ID_ANDROID,
  BUTTON_APP_ID_IOS,
  TEALIUM_ACCOUNT,
  TEALIUM_PROFILE,
  SPINNER_MESSAGE_CHANGE_MS,
  CONNECTION_LOSS_PAUSE_TIMEOUT,
  ONE_TRUST_ANDROID_ID,
  ONE_TRUST_ANDROID_URL,
  ONE_TRUST_IOS_ID,
  ONE_TRUST_IOS_URL,
  MISSION_FEATURED_LIMIT,
  MISSION_SPECIAL_LIMIT,
  MISSION_JUST_FOR_YOU_LIMIT,
  MISSION_TRENDING_LIMIT,
  MISSION_STREAK_LIMIT,
  IMAGE_LOADING_FLICKERING_DEBOUNCE_MS,
  MISSION_SEARCH_HISTORY_LENGTH,
  MISSION_SEARCH_PREDICTIVE_LENGTH,
  PURCHASE_MODAL_TIMEOUT_MS,
  APPSFLYER_DEV_KEY,
  APPSFLYER_APP_ID_ANDROID,
  APPSFLYER_APP_ID_IOS,
  REMOTE_CONFIG_KEY_APP_VERSION,
  REMOTE_CONFIG_CACHE_MS,
  REMOTE_CONFIG_EARN_SECTIONS,
  APP_STORE_URL,
  PLAY_STORE_URL,
  MISSION_TRACKING_TIME_THRESHOLD,
  MISSION_TRACKING_VISIBILITY_THRESHOLD,
  PN_DAYS_TO_REQUEST_PERMISSION,
  LOGIN_REFRESH_INTERVAL_MS,
  POINTS_EXPIRE_BANNER_THRESHOLD_LIST,
  FORTER_SITE_ID,
  STORAGE_CLEAN_WAIT,
  AUTH0_DOMAIN_ID,
  AUTH0_CLIENT_ID,
  AUTH0_AUDIENCE,
  AUTH0_SCOPE,
  TRACK_EVENT_DELAY_MS,
  LOCAL_OFFERS_SEARCH_DEBOUNCE_MS,
  TELLURIDE_API_CONTENT_BASE_URL,
  FORTER_FRAUD_PREVENTION_URL,
  FORTER_DENIED_ACCESS_URL_REGEX,
  FORTER_ALLOWED_ACCESS_URL_REGEX,
  TUTORIAL_HIDE_DELAY_MS,
  TUTORIAL_CONGRATS_DELAY_MS,
  TUTORIAL_END_DELAY_MS,
  TUTORIAL_BANNER_CONGRATS_DELAY_MS,
  TUTORIAL_BANNER_CHECK_DELAY_MS,
  TUTORIAL_BANNER_CHECK_DELAY_FROM_SKIP_MS,
  AB_IOS_NEW_NON_ECM_BANNER_DEV
} from '@env';
const appJson = require('../../app.json'); // tslint:disable-line
const packageJson = require('../../package.json'); // tslint:disable-line
import { FeatureFlag } from '../models/general';
export const ENV = {
  APP_NAME: appJson.displayName,
  APP_VERSION: packageJson.version,
  ENVIRONMENT,
  IS_TEST: ENVIRONMENT === 'test',
  IS_DEV: ENVIRONMENT === 'dev',
  IS_QA: ENVIRONMENT === 'qa',
  IS_UAT: ENVIRONMENT === 'uat',
  IS_PROD: ['uat', 'prod'].includes(ENVIRONMENT),
  BRANCH_NAME,
  SCHEME: SCHEME,
  API: {
    URL: API_URL,
    API_TOKEN_EXPIRATION_PADDING_MS: Number(API_TOKEN_EXPIRATION_PADDING_MS),
    API_RETRY_REQUEST_ATTEMPTS: Number(API_RETRY_REQUEST_ATTEMPTS),
    API_RETRY_REQUEST_ATTEMPT_DELAY_MS: Number(API_RETRY_REQUEST_ATTEMPT_DELAY_MS),
    API_RETRY_REQUEST_STATUS_CODE_RANGES_REGEX: new RegExp(API_RETRY_REQUEST_STATUS_CODE_RANGES_REGEX),
    CACHE: {
      GAME_STATE_MS: Number(API_CACHE_GAME_STATE_MS),
      STREAK_LIST_MS: Number(API_CACHE_STREAK_LIST_MS)
    },
    SLIDE: {
      URL: SLIDE_URL,
      CLIENT_ID: SLIDE_CLIENT_ID,
      CLIENT_SECRET: SLIDE_CLIENT_SECRET,
      CHECKOUT_BASE_URL: SLIDE_CHECKOUT_BASE_URL,
      CHECKOUT_PATH: SLIDE_CHECKOUT_PATH,
      CHECKOUT_CLIENT_HEADER: SLIDE_CHECKOUT_CLIENT_HEADER,
      APPLICATION_ID: SLIDE_APPLICATION_ID,
      FINISHED_URI_PREFIXES: String(SLIDE_APPLICATION_FINISHED_URI_PREFIXES)
    }
  },
  TEALIUM: {
    ACCOUNT: TEALIUM_ACCOUNT,
    PROFILE: TEALIUM_PROFILE
  },
  SPLASH_MIN_ANIMATION_TIME_MS: Number(SPLASH_MIN_ANIMATION_TIME_MS),
  /** @todo move these to a constants file, since these are not configurable */
  STORAGE_KEY: {
    DEVICE_ID: 'device-id-v2',
    TOKEN_SET: 'token-set',
    GAME_LEVEL_UI: 'game-level-user-interaction',
    GAME_CURRENT: 'state-game-current',
    CCPA_SETTING: 'ccpa-setting',
    COOKIE_REGISTRY: 'cookie-registry',
    ONBOARDING: 'onboarding',
    MISSION_SEARCH_HISTORY: 'mission-search-history',
    VIEWED_TOOLTIP_LIST: 'viewed-tooltip-list',
    DISMISSED_TOOLTIP_LIST: 'dismissed-tooltip-list',
    API_OVERRIDE_SETTINGS: 'api-override-settings',
    LOG_SETTINGS: 'log-settings',
    LAST_SUGGESTED_BUILD_VERSION: 'last-suggested-build-version',
    MISSION_AFTER_PURCHASE_MODAL: 'mission-after-purchase-modal',
    LAST_TIME_PN_PERMISSION_ASKED: 'last_time_pn_permission_asked',
    HAS_VISITED_MAIN_SECTION: 'has_visited_main_section',
    EXPIRE_POINTS_BANNER: 'expire-points-banner',
    AUTH0_TOKEN_SET: 'auth0-token-set',
    NEW_ON_MAX_LAST_ONBOARD: 'new-on-max-last-onboard',
    SHOW_OFFERS_ON_EARN_MAIN: 'show-offers-on-earn-main',
    LOCAL_OFFERS_SEARCH_HISTORY: 'local-offers-search-history',
    OFFERS_SEARCH_HISTORY: 'offers-search-history',
    SID: 'sid-tracking-value',
    FORCED_TO_LOGOUT: 'forced-to-logout',
    GIFT_CARDS_SEARCH_HISTORY: 'gc-search-history',
    TUTORIAL_SKIP_STATUS: 'tutorial-skip-status',
    CURRENT_LOCATION: 'current-location',
    RECENTLY_VIEWED_MISSIONS: 'recently-viewed-missions'
  },
  WEBVIEWS: {
    MAX_LOAD_ATTEMPTS: Number(WEBVIEWS_MAX_LOAD_ATTEMPTS),
    ATTEMPT_DELAY_MS: Number(WEBVIEWS_ATTEMPT_DELAY_MS)
  },
  FUSION: {
    FINISHED_URI_REGEX: String(FUSION_APPLICATION_FINISHED_URI_REGEX),
    APPLY_NOW_URI: String(FUSION_APPLY_NOW_WEBVIEW_URI)
  },
  LOCALE,
  IGNORED_FEATURE_LIST: String(IGNORED_FEATURE_LIST)
    .split(',')
    .map<FeatureFlag>(s => s.trim()),
  TOAST_VISIBLE_MS: Number(TOAST_VISIBLE_MS),
  ACTIVITY_FETCH_HISTORY_MONTH_DELTA: Number(ACTIVITY_FETCH_HISTORY_MONTH_DELTA),
  LOCAL_OFFERS: {
    DEFAULT_ACTIVATION_DAYS: Number(LOCAL_OFFERS_DEFAULT_ACTIVATION_DAYS),
    ENROLLMENT_FINISH_URI_REGEX: LOCAL_OFFERS_ENROLLMENT_FINISH_URI_REGEX,
    SEARCH_DEBOUNCE_MS: Number(LOCAL_OFFERS_SEARCH_DEBOUNCE_MS)
  },
  MAX_ACTIVITY_FETCH_HISTORY_MONTH_DELTA: Number(MAX_ACTIVITY_FETCH_HISTORY_MONTH_DELTA),
  MERCHANT: MERCHANT,
  MIN_ACTIVITY_FETCH_HISTORY_RESULTS: Number(MIN_ACTIVITY_FETCH_HISTORY_RESULTS),
  MISSION_LIMIT: {
    FULL: Number(MISSION_FETCH_LIMIT),
    KEEP_EARNING: Number(MISSION_KEEP_EARNING_LIMIT),
    FEATURED: Number(MISSION_FEATURED_LIMIT),
    SPECIAL: Number(MISSION_SPECIAL_LIMIT),
    JUST_FOR_YOU: Number(MISSION_JUST_FOR_YOU_LIMIT),
    TRENDING: Number(MISSION_TRENDING_LIMIT),
    STREAK: Number(MISSION_STREAK_LIMIT)
  },
  CITI: {
    LANDING_URL: CITI_LANDING_URL,
    PHONE_NUMBER: CITI_PHONE_NUMBER
  },
  MODAL_PADDING_MS: Number(MODAL_PADDING_MS),
  ANIMATION: {
    EARN_POINTS: Number(EARN_POINTS_ANIMATION_MS)
  },
  GIFT_CARD_MAX_OPTIONS_COUNT: Number(GIFT_CARD_MAX_OPTIONS_COUNT),
  BUTTON: {
    APP_ID_ANDROID: BUTTON_APP_ID_ANDROID,
    APP_ID_IOS: BUTTON_APP_ID_IOS
  },
  CUSTOM_BRAND_CATEGORY_KEY: 'POPULAR_BRANDS',
  BLACKLIST_BRAND_CATEGORY_KEY: 'BLACKLISTED',
  SPINNER_MESSAGE_CHANGE_MS: Number(SPINNER_MESSAGE_CHANGE_MS),
  SYW_URL: SYW_URL,
  SYW_APP_PROFILE: SYW_APP_PROFILE,
  SYW_SUPPORT_EMAIL: SYW_SUPPORT_EMAIL,
  CONNECTION_LOSS_PAUSE_TIMEOUT: Number(CONNECTION_LOSS_PAUSE_TIMEOUT),
  ONE_TRUST: {
    ANDROID: {
      ID: ONE_TRUST_ANDROID_ID,
      URL: ONE_TRUST_ANDROID_URL
    },
    IOS: {
      ID: ONE_TRUST_IOS_ID,
      URL: ONE_TRUST_IOS_URL
    }
  },
  IMAGE_LOADING_FLICKERING_DEBOUNCE_MS: Number(IMAGE_LOADING_FLICKERING_DEBOUNCE_MS),
  MISSION_SEARCH_HISTORY_LENGTH: Number(MISSION_SEARCH_HISTORY_LENGTH),
  MISSION_SEARCH_PREDICTIVE_LENGTH: Number(MISSION_SEARCH_PREDICTIVE_LENGTH),
  MEMBER_NOT_FOUND_ERROR_CODE: 3053,
  MISSION_MAX_SELECTED_CATEGORIES: 1,
  PURCHASE_MODAL_TIMEOUT_MS: Number(PURCHASE_MODAL_TIMEOUT_MS),
  APPSFLYER: {
    DEV_KEY: APPSFLYER_DEV_KEY,
    APP_ID_ANDROID: APPSFLYER_APP_ID_ANDROID,
    APP_ID_IOS: APPSFLYER_APP_ID_IOS
  },
  REMOTE_CONFIG: {
    DEFAULT_CONFIG: { [REMOTE_CONFIG_KEY_APP_VERSION]: JSON.stringify({ suggested: 0, required: 0 }) },
    CACHE_TIMEOUT: Number(REMOTE_CONFIG_CACHE_MS),
    KEY: {
      APP_VERSION: REMOTE_CONFIG_KEY_APP_VERSION,
      EARN_SECTIONS: REMOTE_CONFIG_EARN_SECTIONS
    }
  },
  STORE_URL: {
    IOS: APP_STORE_URL,
    ANDROID: PLAY_STORE_URL
  },
  MISSION_TRACKING: {
    TIME_THRESHOLD: Number(MISSION_TRACKING_TIME_THRESHOLD),
    VISIBILITY_THRESHOLD: Number(MISSION_TRACKING_VISIBILITY_THRESHOLD)
  },
  PN_DAYS_TO_REQUEST_PERMISSION: Number(PN_DAYS_TO_REQUEST_PERMISSION),
  LOGIN_REFRESH_INTERVAL_MS: Number(LOGIN_REFRESH_INTERVAL_MS),
  POINTS_EXPIRE_BANNER_THRESHOLD_LIST: POINTS_EXPIRE_BANNER_THRESHOLD_LIST.split(',').map(Number) as number[],
  ROUTE_HISTORY_MAX_LENGTH: 50,
  FORTER_SITE_ID: FORTER_SITE_ID,
  STORAGE_CLEAN_WAIT: Number(STORAGE_CLEAN_WAIT),
  AUTH0: {
    DOMAIN_ID: AUTH0_DOMAIN_ID,
    CLIENT_ID: AUTH0_CLIENT_ID,
    AUDIENCE: AUTH0_AUDIENCE,
    SCOPE: AUTH0_SCOPE
  },
  SAILTHRU_MESSAGES_CATEGORIES: {
    NEW_ON_MAX: 'NEW_ON_MAX'
  },
  TRACK_EVENT_DELAY_MS: Number(TRACK_EVENT_DELAY_MS),
  TELLURIDE_API: {
    CONTENT_BASE_URL: TELLURIDE_API_CONTENT_BASE_URL
  },
  INACTIVE_DURATION_LIMIT_TIME: 5,
  FORTER_API: {
    FRAUD_PREVENTION_URL: FORTER_FRAUD_PREVENTION_URL,
    DENIED_ACCESS_URL_REGEX: FORTER_DENIED_ACCESS_URL_REGEX,
    ALLOWED_ACCESS_URL_REGEX: FORTER_ALLOWED_ACCESS_URL_REGEX
  },
  TUTORIAL: {
    HIDE_DELAY_MS: TUTORIAL_HIDE_DELAY_MS,
    CONGRATS_DELAY_MS: TUTORIAL_CONGRATS_DELAY_MS,
    END_DELAY_MS: TUTORIAL_END_DELAY_MS,
    BANNER: {
      CONGRATS_DELAY_MS: TUTORIAL_BANNER_CONGRATS_DELAY_MS,
      CHECK_DELAY_MS: TUTORIAL_BANNER_CHECK_DELAY_MS,
      CHECK_DELAY_FROM_SKIP_MS: TUTORIAL_BANNER_CHECK_DELAY_FROM_SKIP_MS
    }
  },
  EXPERIMENTS: {
    AB_IOS_NEW_NON_ECM_BANNER_DEV: AB_IOS_NEW_NON_ECM_BANNER_DEV
  }
};
