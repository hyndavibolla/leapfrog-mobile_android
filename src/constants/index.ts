export * from './env';
export * from './routes';
export * from './eventTracking';
export * from './styles';
export * from './icons';
export * from './constants';
