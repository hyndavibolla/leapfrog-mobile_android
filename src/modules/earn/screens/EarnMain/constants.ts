export const EarnSections = {
  NEW_ON_MAX: 'newOnMax',
  FEATURED_PARTNERS: 'featuredPartners',
  RECENTLY_VIEWED_MISSIONS: 'recentlyViewedMissions',
  ACTIVE_MISSION_OFFERS: 'activeMissionOffers',
  TOP_CATEGORIES: 'topCategories',
  TOP_BRANDS: 'topBrands',
  SURVEYS: 'surveys',
  FEATURED_MISSIONS: 'featuredMissions',
  CLAIM_YOUR_REWARDS: 'claimYourRewards',
  MAP_CARD: 'mapCard'
};

export const EarnSectionsData = {
  sections: [
    {
      name: EarnSections.NEW_ON_MAX,
      order: 0,
      visible: true,
      title: 'New on MAX',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.FEATURED_PARTNERS,
      order: 1,
      visible: true,
      title: 'Featured Partners',
      description: '',
      seeAllButton: false
    },
    {
      name: EarnSections.RECENTLY_VIEWED_MISSIONS,
      order: 2,
      visible: true,
      title: 'Recently viewed brands',
      description: '',
      seeAllButton: false
    },
    {
      name: EarnSections.ACTIVE_MISSION_OFFERS,
      order: 3,
      visible: true,
      title: 'ðŸ”¥ Mission Offers',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.MAP_CARD,
      order: 4,
      visible: true,
      title: '',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.TOP_CATEGORIES,
      order: 5,
      visible: true,
      title: 'Top Categories',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.TOP_BRANDS,
      order: 6,
      visible: true,
      title: 'Top Brands',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.SURVEYS,
      order: 7,
      visible: true,
      title: 'Surveys',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.FEATURED_MISSIONS,
      order: 8,
      visible: true,
      title: 'Featured Missions',
      description: '',
      seeAllButton: true
    },
    {
      name: EarnSections.CLAIM_YOUR_REWARDS,
      order: 9,
      visible: true,
      title: 'Claim your Rewards!',
      description: '',
      seeAllButton: true
    }
  ]
};
