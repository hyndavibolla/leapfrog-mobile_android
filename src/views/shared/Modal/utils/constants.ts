export enum ModalType {
  CENTER = 'center',
  BOTTOM = 'bottom'
}

export enum ModalSize {
  SMALL = 'small',
  MEDIUM = 'medium',
  EXTRA_LARGE = 'extralarge',
  DYNAMIC = 'dynamic',
  FULL_SCREEN = 'fullscreen'
}

export const ModalSizeToHeightMap = {
  [ModalSize.SMALL]: 40,
  [ModalSize.MEDIUM]: 50,
  [ModalSize.EXTRA_LARGE]: 80,
  [ModalSize.FULL_SCREEN]: 100
};
