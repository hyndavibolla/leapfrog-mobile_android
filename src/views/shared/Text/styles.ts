import styled from '@emotion/native';

import { FONT_FAMILY, COLOR, FONT_SIZE, LINE_HEIGHT } from '_constants';

import { Props } from './Text';

export const StyledText = styled.Text<Props>`
  font-family: ${({ font }: Props) => font || FONT_FAMILY.MEDIUM};
  color: ${({ color }: Props) => color || COLOR.DARK_GRAY};
  font-size: ${({ size }: Props) => size || FONT_SIZE.SMALLER};
  line-height: ${({ lineHeight }: Props) => lineHeight || LINE_HEIGHT.PARAGRAPH};
`;
