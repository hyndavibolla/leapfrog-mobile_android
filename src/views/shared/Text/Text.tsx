import React, { memo, ReactNode } from 'react';
import { TextProps } from 'react-native';

import { COLOR, FONT_FAMILY, FONT_SIZE, LINE_HEIGHT } from '_constants';

import { StyledText } from './styles';

export interface Props extends TextProps {
  color?: COLOR;
  font?: FONT_FAMILY;
  size?: FONT_SIZE;
  lineHeight?: LINE_HEIGHT;
  children: ReactNode;
}

const Text = ({ color, font, size, lineHeight, children, ...otherProps }: Props) => {
  return (
    <StyledText color={color} font={font} size={size} lineHeight={lineHeight} {...otherProps}>
      {children}
    </StyledText>
  );
};

export default memo(Text);
