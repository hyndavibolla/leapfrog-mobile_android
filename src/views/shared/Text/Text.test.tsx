import React from 'react';
import { render } from '@testing-library/react-native';

import Text from './Text';
import { FONT_FAMILY } from '../../../constants';

describe('Text', () => {
  it('should render with default font', () => {
    const { toJSON } = render(<Text>default font</Text>);
    expect(toJSON()).toMatchSnapshot();
  });

  it('should render with custom font', () => {
    const { toJSON } = render(<Text font={FONT_FAMILY.BOLD}>custom font</Text>);
    expect(toJSON()).toMatchSnapshot();
  });
});
