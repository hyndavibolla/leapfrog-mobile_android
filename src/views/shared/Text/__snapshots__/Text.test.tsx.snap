// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should render with custom font 1`] = `
<Text
  font="SFProDisplay-Bold"
  style={
    Array [
      Object {
        "color": "#909BA0",
        "fontFamily": "SFProDisplay-Bold",
        "fontSize": 16,
        "lineHeight": 22,
      },
      undefined,
    ]
  }
>
  custom font
</Text>
`;

exports[`Text should render with default font 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#909BA0",
        "fontFamily": "SFProDisplay-Medium",
        "fontSize": 16,
        "lineHeight": 22,
      },
      undefined,
    ]
  }
>
  default font
</Text>
`;
